### PR TITLE
[Do not merge] Govuk Release app with PostgreSQL backend

### PR DESCRIPTION
**Pull request by djsd123**
What
Modifying the release app to use Postgres instead of MySql.
Automate release user provisioning.
Allow startup.sh to use unicorn by default.

How
Swapping the MySql adapter for Postgres adapter. Reflected change in database yaml config.
Created Rake task to provision release app user.
Added handler gem to gemfile. Specified Unicorn argument to command.

### CODE REVIEW

File: lib/tasks/postgres.rake [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'pg'
+require 'yaml'
+
+namespace :pg do 
+
+	def conf
```
Comment (boffbowsh) [inline lib/tasks/postgres.rake]: We use 2 spaces for indentation in Ruby

**Comment by boffbowsh** [comment]
This will break our current install. Is there a plan for migrating the data?

File: lib/tasks/postgres.rake [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	def conf 
+		@conf ||= YAML.load_file("#{Rails.root}/config/database.yml")
+	end
+
+	def username
+		@username ||= conf['development']['username']
+	end
+
+	desc "Create release user and load schemas"
+	task :Create_everything do
```
Comment (boffbowsh) [inline lib/tasks/postgres.rake]: rake tasks should be all lower case

**Comment by alexmuller** [comment]
> Is there a plan for migrating the data?

Yep, we know it's a thing that needs to happen.

**Comment by djsd123** [comment]
Thanks @boffbowsh @alexmuller 
Yep, I'm just pushing my changes as I go. Added WIP to title. 

I was thinking in terms migrating data I could
- dump data to a bucket (If it's not being backed up to one already)
- automate an import from the bucket as an initial provisioning task along with creating accounts.

**Comment by djsd123** [comment]
To be reviewed by @boffbowsh

**Comment by boffbowsh** [comment]
I've marked this as *Do not merge* because it will break deployment of the release app on our current infrastructure. What we've done in the past for app migrations is enable compatibility with both DBs: https://github.com/alphagov/support-api/pull/63. This might be overkill for a small app such as this though, so I'd also support deploying this to the PaaS from a branch. But I don't think it should be merged into master until we're ready to put it into production.

File: lib/tasks/postgres.rake [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  task :get_data do
+    data = Aws::S3::Resource.new(
+      region: 'eu-west-1',
+      access_key_id: ENV['AWS_ACCESS_KEY_ID'],
+      secret_access_key: ENV['AWS_SECRET_ACCESS_KEY'],
+    )
+
+    data.bucket('govuk-release-app-test').object('postgresql_backend/release_app_data.tar').get(response_target: './response.txt')
```
Comment (boffbowsh) [inline lib/tasks/postgres.rake]: Can we include the import script in this PR too? At the moment I don't understand how this line allows us to import the data into the PaaS

Comment (djsd123) [inline lib/tasks/postgres.rake]: Hi @boffbowsh,
It doesn't. It just pulls data from a bucket.
When I spoke to @alexmuller last he said an import script would not be necessary and to do it manually as a one off.  I'll prompt him to read your comments when he's free.

Thanks.

Comment (boffbowsh) [inline lib/tasks/postgres.rake]: This task will also be a one off though, so I don't understand why this is included in the PR but the one off import isn't. I'd like to actually review how you're planning to import the data, currently I don't know what's in that tarball and how you're going to load that into the new environment.

**Comment by boffbowsh** [comment]
Can you squash the commits please?

**Comment by alexmuller** [comment]
Let's close this while we think about secrets and deployment and come back to it in a bit.

**Pull request closed without merging**